draft-ietf-tls-dtls-connection-id.md

```diff
@@ -134,44 +133,67 @@ indicates that the server will send with the client's CID but does not
+receive as a CID in encrypted records, it is possible
 for an endpoint to use a globally constant length for such connection
 identifiers.  This can in turn ease parsing and connection lookup,
-
 for example by having the length in question be a compile-time constant.
 Implementations, which want to use variable-length CIDs, are responsible
 for constructing the CID in such a way that its length can be determined
-on reception. Note that such implementations must still be able to send
-connection identifiers of different length to other parties.
-
-Note that it is not possible to parse the records without knowing how 
-long the CID is.
+on reception. Such implementations must still be able to send
+CIDs of different length to other parties. Note that there is no CID 
+length information included in the record itself.
 
 In DTLS 1.2, CIDs are exchanged at the beginning of the DTLS
 session only. There is no dedicated "CID update" message
 that allows new CIDs to be established mid-session, because
 DTLS 1.2 in general does not allow TLS 1.3-style post-handshake messages
-that do not themselves begin other handshakes. 
+that do not themselves begin other handshakes. When a DTLS session is 
+resumed or renegotiated, the "connection_id" extension is negotiated afresh. 
+
+If DTLS peers have not negotiated the use of CIDs then the RFC 6347-defined 
+record format and content type MUST be used. 
+
+If DTLS peers have negotiated the use of a CIDs using the ClientHello and
+the ServerHello messages then the peers need to take the following steps.
 
-DTLS peers switch to the new record layer format, i.e., the record layer format 
-containing the CID, when encryption is enabled.
+The DTLS peers determine whether incoming and outgoing messages need 
+to use the new record format, i.e., the record format containing the CID. 
+The new record format and the CID content type is only used once encryption 
+is enabled. Plaintext payloads never use the new record type and the CID content 
+type. 
 
-# Record Layer Extensions and Record Payload Protection
+For sending, if a zero-length CID has been negotiated then the RFC 6347-defined 
+record format and content type MUST be used (see Section 4.1 of {{RFC6347}})
+else the new record layer format and the new content type defined in {{dtls-record12}} MUST be used. 
+
+When transmitting a datagram with the new record format and the new content type, 
+the new MAC computation defined in {{mac}} MUST be used.
+
+For receiving, if the tls12_cid content type is set, then the CID is used to look up 
+the connection and the security association. If the tls12_cid content type is not set, 
+then the connection and security association is looked up by the 5-tuple and a 
+check MUST be made to determine whether the expected CID value is indeed 
+zero length. If the check fails, then the datagram MUST be dropped. 
+
+When receiving a datagram with the new record format and the new content type, 
```

Comment:
When reading this condition one might assume that it's possible to receive a datagram with the new record format but without the new content type. Should we instead fold this together and say that (a) if the new content type is used then the new record format MUST also be in effect, and (b) if the new content type is not used then the old record format MUST be used?

Reply:
I tried to change the wording to make it more clear that the new record format and the new content type is actually the same. There is no new record format that does not contain the new content type. 

Have a look at the new text